akka-remote/src/main/scala/akka/remote/artery/BufferPool.scala

```diff
@@ -366,65 +404,86 @@ private[remote] final class EnvelopeBuffer(val byteBuffer: ByteBuffer) {
 
     // Write fixed length parts
     byteBuffer.put(VersionOffset, header.version)
-    byteBuffer.put(FlagsOffset, header.flags)
+
+    // frameLength and streamId were added in version 1
+    if (header.version >= 1) {
+      byteBuffer.putInt(FrameLengthOffset, header.frameLength)
+      byteBuffer.putInt(StreamIdOffset, header.streamId)
+    }
+    val adjust = header.versionPositionAdjust
```

Comment:
Alternatively we could use the relative methods then position the bytebuffer at the end to where ever it needs to be, which looks like at the MetadataContainerAndLiteralSectionOffset

Reply:
yeah, it was rather difficult to introduce this non-compatible header format change. I guess it's the price we have to pay for not using a serialization library for the header. At least we have the version number field. If we have to do more bigger changes later we might want to do something different. I think this adjust offset works for this small change.